src/analysis/seminorm.lean

```diff
@@ -301,6 +301,9 @@ begin
         nnreal.coe_max, subtype.coe_mk, ih] }
 end
 
+lemma le_finset_sup_apply {p : ι → seminorm 𝕜 E} {s : finset ι} {i : ι} (hi : i ∈ s) {x : E} :
+  p i x ≤ s.sup p x := (finset.le_sup hi : p i ≤ s.sup p) x
```

Comment:
I think an `apply_le_of_le` lemma would be more useful here. Then you don't need this lemma.

Reply:
[`pi.le_def`](https://leanprover-community.github.io/mathlib_docs/order/basic.html#pi.le_def) ?

Reply:
`pi.le_def` chained with the local `le_def`.

Though trying this out, perhaps the real problem is that `finset.le_sup` should take the `f` argument explicitly.

Reply:
```suggestion
lemma le_def' {f g : seminorm 𝕜 E} : f ≤ g ↔ ∀ x, f x ≤ g x := iff.rfl
```
This proof then just becomes `le_def'.mp (finset.le_sup hi) x` which can be inlined where needed.

Reply:
I will make a separate PR with the correct renaming. It is probably better to do this, before the port hits `Analysis.Seminorm`.

Reply:
I would recommend adding a comment to https://github.com/leanprover-community/mathlib4/wiki/port-comments to ensure that the port does not hit Analysis.Seminorm before you are ready.

Reply:
I've actually tried today to port it, but it is quite messy. I've added the message.